src/components/AppIcon/AppIcon.js

```diff
@@ -0,0 +1,88 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import { appIconUrl } from '../../utils'
+import RemoteImage from '../RemoteImage'
+
+import iconSvgHome from './assets/app-home.svg'
+import iconSvgDefault from './assets/app-default.svg'
+
+const DEFAULT_SIZE = 22
+const DEFAULT_RADIUS = 5
+
+const AppIcon = ({ app, src, size, radius, ...props }) => {
+  if (radius === -1) {
+    radius = size * (DEFAULT_RADIUS / DEFAULT_SIZE)
+  }
+  return (
+    <div
+      css={`
+        display: flex;
+        align-items: center;
+        overflow: hidden;
+        border-radius: ${radius}px;
+      `}
+      {...props}
+    >
+      {(() => {
```

Comment:
Wow!
Would it have been cleaner to use another component instead of a self executing function?

Reply:
Sure, I don’t have any preference for one or the other (until we have [pattern matching](https://github.com/tc39/proposal-pattern-matching) 💥), I tend to use this pattern because I find it more readable than using nested ternary operators, and declaring + calling a function at render time is something we do a lot with render props, so I don’t think it is an issue performance wise, as long as we don’t render at 60 times per second.

Having internal components is nice too, but the problem is that ESLint insists that every component requires prop types, which makes it super verbose. I couldn’t find any setting to only apply this rule to exported components.

Maybe we could directly call functions for this? That would also skip the extra step of creating a React element + having React calling it (insignificant gain in most cases, but it can be an argument to decide).

```jsx
function MyComponent(props) {
  return (
    <div>
      {myComponentContent(props)}
    </div>
  )
}

function myComponentContent({ a, b, c }) {
  if (a) {
    return <span>{a}</span>
  }
  return <span>default</span>
}
```

Another solution would be to disable this rule after the main component, which would make it possible to have light internal components:

```jsx
function MyComponent(props) {
  return (
    <div>
      <MyComponentContent {...props} />
    </div>
  )
}

MyComponent.propTypes = { /* … */ }

// Disabling the ESLint prop-types check for internal components.
/* eslint-disable react/prop-types */

function MyComponentContent({ a, b, c }) {
  if (a) {
    return <span>{a}</span>
  }
  return <span>default</span>
}
```

And for simple cases, I think the “IIFE in JSX” pattern, while not being the most elegant thing in the world, can do a good job too 😆

```jsx
function MyComponent(props) {
  return (
    <div>
      {(() => {
        if (a) {
          return <span>{a}</span>
        }
        return <span>default</span>
      })()}
    </div>
  )
}
```


@AquiGorka @sohkai I don’t really have any preference, I think the three solutions are fine and I use them depending on the context. I pushed `#2` for now as @AquiGorka was suggesting a preference for it rather than `#3`. Let me know if you have any opinion about it so we align!

Reply:
❤️  @bpierre thank you for taking the time to explain your line of thinking and yes I prefer `#2` but it is only a personal preference and I can adhere to any team convention we agree on - and yes, because of `prop types` it becomes a bit too verbose.

Reply:
Cool :-) Let’s not make it a convention for now then, and we can refer to this discussion if we want to avoid one of these patterns in the future!